src/mixins/Projector.ts

```diff
@@ -304,18 +304,13 @@ export function ProjectorMixin<P, T extends Constructor<WidgetBase<P>>>(Base: T)
 				this.setProperties(this._projectorProperties);
 			}
 			let result = super.__render__();
-			if (typeof result === 'string' || result === null) {
+
+			if (Array.isArray(result) || typeof result === 'string' || result === null) {
 				if (!this._rootTagName) {
 					this._rootTagName = 'span';
 				}
 
-				result = {
-					domNode: null,
-					vnodeSelector: this._rootTagName,
-					properties: {},
-					children: undefined,
-					text: result === null ? undefined : result
-				};
+				result = h(this._rootTagName, {}, result);
```

Comment:
turns out `h()` is good for creating `VNode`s and will do all the flattening and filtering needed 😄